sw/device/lib/testing/hmac_testutils.h

```diff
@@ -0,0 +1,85 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_SW_DEVICE_LIB_TESTING_HMAC_TESTUTILS_H_
+#define OPENTITAN_SW_DEVICE_LIB_TESTING_HMAC_TESTUTILS_H_
+
+#include <assert.h>
+#include <stdint.h>
+
+#include "sw/device/lib/dif/dif_hmac.h"
+
+// Tests need to understand the byte order of the data in the string and
+// the digest values below, as they are laid out for the current processor.
+// RISC-V is little-endian, so the first of these `kHmacTransactionConfig`
+// values is the one we expect to be used, but we include the other for
+// completeness.
+#define HMAC_TESTUTILS_ASSERT_LITTLE_ENDIAN                \
+  static_assert(__BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__, \
+                "This test assumes the target platform is little endian.")
+/**
+ * FIFO empty 10uS timeout, which should be enough even for a 100kHz device.
+ *
+ * https://docs.opentitan.org/hw/ip/hmac/doc/
+ * Single HMAC block compression takes 80 cycles.
+ */
+#define HMAC_TESTUTILS_FIFO_EMPTY_TIMEOUT_USEC 10
+
+/**
+ * HMAC done 50uS timeout, which should be enough even for a 100kHz device.
+ *
+ * https://docs.opentitan.org/hw/ip/hmac/doc/
+ * Final hash calculation takes 360 cycles, which consists of one
+ * block compression and extra HMAC computation.
+ */
+#define HMAC_TESTUTILS_FINISH_TIMEOUT_USEC 50
+
+/**
+ * Reads and compares the actual length sent message length against expected.
+ *
+ * The message length is provided in bits.
+ */
+void hmac_testutils_check_message_length(const dif_hmac_t *hmac,
+                                         uint64_t expected_sent_bits);
```

Comment:
Why is this a u64 rather than a size_t, and why is it in bits? There's no non-byte-aligned digests that we support.

Reply:
Because HMAC DIF returns u64.